GoogleForCreators/web-stories-wp
# Perf: Improve selection latency by reducing renders

## Context

Just more noodling on element selection performance.

## Summary

According to React profiler, reduces cost of text element selection from ~280 to ~100ms. 

- Memoize renders of display elements
- Memoize renders of sections of UI that don't change on element selection: carousel container, header, page menu
- Skip render of contents in design panels when collapsed

Flamegraphs from the first commit following text element selection click:

### Before
![Screen Shot 2021-07-12 at 6 40 05 PM](https://user-images.githubusercontent.com/20525523/125365961-310a3a80-e343-11eb-813a-57c9f8558a8b.png)

### After
![Screen Shot 2021-07-12 at 6 42 31 PM](https://user-images.githubusercontent.com/20525523/125365977-36678500-e343-11eb-9c20-5c476171f70a.png)

## Relevant Technical Choices

N/A

## To-do

- Checklist UI components are still rerendering on selection but should be fixed with #8289

## User-facing changes

- Changing element selection should be more responsive

## Testing Instructions



### QA



This PR can be tested by following these steps:

1. Select a text element in the editor
2. The editor should update slightly faster (display selection box & update text panel)
3. The rest of the editor should work normally

### UAT






## Reviews

### Does this PR have a security-related impact?

No.

### Does this PR change what data or activity we track or use?

No.

### Does this PR have a legal-related impact?

No.

## Checklist




---



Partial for #8259
